controllers/common.go

```diff
@@ -59,22 +60,23 @@ func BuildCollector(params manifests.Params) ([]client.Object, error) {
 		}
 		resources = append(resources, objs...)
 	}
-	// TODO: Remove this after TargetAllocator CRD is reconciled
-	if params.TargetAllocator != nil {
-		taParams := targetallocator.Params{
-			Client:          params.Client,
-			Scheme:          params.Scheme,
-			Recorder:        params.Recorder,
-			Log:             params.Log,
-			Config:          params.Config,
-			Collector:       &params.OtelCol,
-			TargetAllocator: *params.TargetAllocator,
-		}
-		taResources, err := BuildTargetAllocator(taParams)
-		if err != nil {
-			return nil, err
+	if !featuregate.CollectorUsesTargetAllocatorCR.IsEnabled() {
```

Comment:
can you add a comment here specifying why this is negated?

Reply:
Sure. It's because here we have the collector builder also build the TA manifests. It should only do that if the flag is disabled; otherwise it should just build the TargetAllocator CR, which happens in the normal path.

Reply:
yep, understood. this is more for future committers incase they don't see that.

Reply:
I expect to get rid of this gate pretty quickly. It's really only here to make it easier for review and merge these changes without enabling anything by default for users.